Mathlib/LinearAlgebra/Matrix/GeneralLinearGroup.lean

```diff
@@ -329,4 +331,85 @@ def planeConformalMatrix {R} [Field R] (a b : R) (hab : a ^ 2 + b ^ 2 ≠ 0) :
+    Fin.tail_cons, heq_eq_eq, and_self]
+
+lemma card_LinearInependent_subtype {k : ℕ} (hk : k ≤ n) :
+    Nat.card { s : Fin k → (Fin n → 𝔽) // LinearIndependent 𝔽 s } =
+      ∏ i : Fin k, ((q) ^ n - (q) ^ i.val) := by
+  have {m : ℕ} : Fintype ({ s : Fin m → (Fin n → 𝔽) // LinearIndependent 𝔽 s}) :=
+      Fintype.ofFinite _
+  rw [Nat.card_eq_fintype_card]
+  induction' k with k ih
+  · simp only [LinearIndependent, Finsupp.total_fin_zero, ker_zero, Fintype.card_ofSubsingleton,
+    Finset.univ_eq_empty, Finset.prod_empty]
+  · have {s : { s : Fin k → Fin n → 𝔽 // LinearIndependent 𝔽 s }} :
+      Fintype (Submodule.span 𝔽 (Set.range (s : Fin k → Fin n → 𝔽)) : Set (Fin n → 𝔽)) :=
+    Fintype.ofFinite _
+    have {s : { s : Fin k → Fin n → 𝔽 // LinearIndependent 𝔽 s }} :
+        Fintype ((Submodule.span 𝔽 (Set.range (s : Fin k → Fin n → 𝔽)))ᶜ : Set (Fin n → 𝔽)) :=
+      Fintype.ofFinite _
+    have (s : { s : Fin k → Fin n → 𝔽 // LinearIndependent 𝔽 s }) :
+        Fintype.card ((Submodule.span 𝔽 (Set.range (s : Fin k → Fin n → 𝔽)))ᶜ : Set (Fin n → 𝔽)) =
+      (q) ^ n - (q) ^ k := by
+        simp only [Fintype.card_compl_set, Fintype.card_pi, Finset.prod_const, Finset.card_univ,
+          Fintype.card_fin, SetLike.coe_sort_coe,
+          card_eq_pow_finrank (K := 𝔽) (V := Submodule.span 𝔽 (Set.range (s : Fin k → Fin n → 𝔽))),
+          finrank_span_eq_card s.property]
+    simp only [Fintype.card_congr (inductiveStepEquiv n), Fintype.card_sigma,
+      Fintype.sum_congr _ _ this, Finset.sum_const, Finset.card_univ, ih (Nat.le_of_succ_le hk),
+      smul_eq_mul, mul_comm, Fin.prod_univ_succAbove _ k, Fin.natCast_eq_last, Fin.val_last,
+      Fin.succAbove_last, Fin.coe_castSucc]
+
+/-- Equivalence between `GL n F` and `n` vectors of length `n` that are linearly independent. Given
+by sending a matrix to its coloumns. -/
+noncomputable def equiv_GL_linearindependent {F : Type*} [Field F] (hn : 0 < n) :
+    GL (Fin n) F ≃ { s : Fin n → (Fin n → F) // LinearIndependent F s } where
+  toFun M := ⟨transpose M, by
+    apply linearIndependent_iff_card_eq_finrank_span.2
+    rw [Set.finrank, ← rank_eq_finrank_span_cols, rank_unit]⟩
+  invFun M := by
+    apply GeneralLinearGroup.mk'' (transpose (M.1))
+    rw [show M.1ᵀ = Basis.toMatrix (Pi.basisFun F (Fin n)) (transpose (M.1)ᵀ) by rfl,
+      transpose_transpose]
+    have : Nonempty (Fin n) := Fin.pos_iff_nonempty.1 hn
+    have hdim : Fintype.card (Fin n) = FiniteDimensional.finrank F (Fin n → F) := by
+      simp only [Fintype.card_fin, FiniteDimensional.finrank_fintype_fun_eq_card]
+    let b := basisOfLinearIndependentOfCardEqFinrank M.2 hdim
+    rw [show M = ⇑b by simp only [b, coe_basisOfLinearIndependentOfCardEqFinrank]]
+    have : Invertible ((Pi.basisFun F (Fin n)).toMatrix ⇑b) :=
+      (Pi.basisFun F (Fin n)).invertibleToMatrix b
+    exact isUnit_det_of_invertible _
+  left_inv := fun x ↦ Units.ext (ext fun i j ↦ rfl)
+  right_inv := by exact congrFun rfl
```

Comment:
```suggestion
noncomputable def equiv_GL_linearindependent {F : Type*} [Field F] (hn : 0 < n) :
    GL (Fin n) F ≃ { s : Fin n → (Fin n → F) // LinearIndependent F s } where
  toFun M := ⟨transpose M, by
    apply linearIndependent_iff_card_eq_finrank_span.2
    rw [Set.finrank, ← rank_eq_finrank_span_cols, rank_unit]⟩
  invFun M := GeneralLinearGroup.mk'' (transpose (M.1)) <| by
    have : Nonempty (Fin n) := Fin.pos_iff_nonempty.1 hn
    rw [← Basis.coePiBasisFun.toMatrix_eq_transpose,
      ← coe_basisOfLinearIndependentOfCardEqFinrank M.2]
    let b := basisOfLinearIndependentOfCardEqFinrank M.2 (by simp)
    have := (Pi.basisFun F (Fin n)).invertibleToMatrix b
    exact isUnit_det_of_invertible _
  left_inv := fun x ↦ Units.ext (ext fun i j ↦ rfl)
  right_inv := by exact congrFun rfl
```
In general it is better to avoid `rw [show ... by rfl]`: if there is a lemma you can rewrite just use it, otherwise you can add it. The reason is that if at some point someone change the definitions involved your proof will break and it will difficult to maintain (on the other hand, if you rewrite a lemma it will be that lemma that will broke and it is usually easier to fix).

Reply:
Well, it's even already there!